packages/playground/blueprints/src/lib/steps/activate-plugin.ts

```diff
@@ -65,22 +64,61 @@ export const activatePlugin: StepHandler<ActivatePluginStep> = async (
 				}
 			}
 
-			if ( null === $response ) {
-				die('Plugin activated successfully');
-			} else if ( is_wp_error( $response ) ) {
-				throw new Exception( $response->get_error_message() );
+			if (is_wp_error($response)) {
+				die($response->get_error_message());
 			}
+		`,
+	});
+
+	/**
+	 * Instead of checking the plugin activation response,
+	 * check if the plugin is active by looking at the active plugins list.
+	 *
+	 * We have to split the activation and the check into two PHP runs
+	 * because some plugins might redirect during activation,
+	 * which would prevent any output that happens after activation from being returned.
+	 *
+	 * Relying on the plugin activation response is not reliable because if the plugin activation
+	 * produces any output, it will be threaded as an error.
+	 * See WordPress source code for more details:
+	 * https://github.com/WordPress/wordpress-develop/blob/6.7/src/wp-admin/includes/plugin.php#L733
+	 *
+	 * Because some plugins can create an output, we need to use output buffering
+	 * to ensure the 'true' response is not polluted by other outputs.
+	 * If the plugin activation fails, we will return the buffered output as it might
+	 * contain more information about the failure.
+	 */
+	const isActiveCheckResult = await playground.run({
+		code: `<?php
+			ob_start();
+			require_once( ${phpVar(docroot)}. "/wp-load.php" );
 
-			throw new Exception( 'Unable to activate plugin' );
+			$relative_plugin_path = str_replace( ${phpVar(
+				docroot
+			)}. "/wp-content/plugins/", '', ${phpVar(pluginPath)} );
+			$active_plugins = get_option( 'active_plugins' );
+			foreach ( $active_plugins as $plugin ) {
+				if ( strpos( $plugin, $relative_plugin_path ) === 0 ) {
```

Comment:
This would yield a false positive if one plugin is called `gutenberg` and another one is called `gutenberg-beta`. Let's account for that.

Reply:
How would this happen? We required absolute or relative plugin paths for plugin activation, not slugs.

`gutenberg/gutenberg.php` wouldn't result in true if `gutenberg-beta/guternberg.php` was active.

Reply:
My suggestion for the whole section is to be more explicit on why you use `str_replace` vs `substr` or `strpos` vs `===`. The former ones can be considered "fuzzy" while the latter ones are more precise. We should opt for the latter if possible and if not, explain why this is needed in a comment.

Reply:
I have a working solution, but it's not clean. Let me think a bit more about it.

Reply:
I update the is active check to either match the full `PLUGIN_DIR/PLUGIN.php` pattern or `PLUGIN_DIR/`. 
If it's a `PLUGIN_DIR/` the trailing slash will ensure we are checking the full plugin directory name and not a partial name. 

Let me know what you think of the implementation https://github.com/WordPress/wordpress-playground/pull/2066/files#diff-4a02b123bb6311fff3ec089ca1ade75c75f1af35aaa4f1a717f14d444fc9b485R101-R147

Reply:
This is looking really good, thank you @bgrgicak! I left a few minor nitpicks, no major changes needed.